File: Greedy-Algo-books/number_card_game.cpp
#include<iostream>
#include<algorithm>
#include<vector>
#include<map>
#include<unordered_map>
#include<ctime>
using namespace std;

int main()
{
	//check time
	time_t start_time = time(0);

	int N, M, K;
	int min = 10000;
	int answer = 0;
	cin >> N >> M;


	unordered_map<int, int> minimumMap;

	for (int i = 0; i < N; i++)
	{
		min = 10000;

		for (int j = 0; j < M; j++)
		{
			cin >> K;
			if (min > K) min = K;
		}

		minimumMap.insert(make_pair(i, min));
	}

	//min을 기준으로  내림 차순으로 제일 큰 값을 구하기
	int max = -1;
	for (pair<int, int> minimum : minimumMap)
	{
		if (max < minimum.second)
			answer = minimum.second;
	}

	cout << answer << endl;
	time_t end_time = time(0);

	cout << "time : " << end_time - start_time;

	return 0;
}

File: python/greedy_algorithms/number_card_game.py
# #시간 계산
# import time
# start_time = time.time()
# 입력 받기
N, M = map(int, input().split())

answer = 0

# g한 줄씩 입력 받아 확인
for i in range(N):
    # 입력
    data = list(map(int, input().split()))

    # min 값 구하기
    minimum = min(data)
    # 혹은 이중 포문으로 두개만 비교해서 최솟값 구하기

    # minimum값에서 가장 큰 값 answer에 저장
    answer = max(answer, minimum)


print(answer)

# end_time = time.time()
# print("time : " , end_time - start_time)
